.github/workflows/tox.yml

```diff
@@ -1,11 +1,20 @@
-name: tox
+name: ruff_and_tox
```

Comment:
I’m wondering whether we should split this into separate code quality and tests files since there really isn’t a need to run Ruff as a blocker for tox, but it also isn’t that important.

Reply:
If codespell + ruff took more than 10 seconds then I would not run them as a precursor but given that they are so fast that it is a painless precaution that often saves time and energy.

Reply:
Yeah, I think the job setup & teardown takes longer than the actual ruff execution.